p751toolbox/field_arm64.s

```diff
@@ -0,0 +1,2408 @@
+	LDP	32(R1), (R15, R16)
+	LDP	48(R0), (R9, R10)
+	LDP	48(R1), (R17, R19)
+	ADCS	R15, R7
+	ADCS	R16, R8
+	STP	(R7, R8), 32(R2)
+	ADCS	R17, R9
+	ADCS	R19, R10
+	STP	(R9, R10), 48(R2)
+
+	LDP	64(R0), (R11, R12)
+	LDP	64(R1), (R15, R16)
+	LDP	80(R0), (R13, R14)
+	LDP	80(R1), (R17, R19)
+	ADCS	R15, R11
+	ADCS	R16, R12
+	STP	(R11, R12), 64(R2)
+	ADCS	R17, R13
+	ADCS	R19, R14
+	STP	(R13, R14), 80(R2)
+
+	LDP	96(R0), (R3, R4)
+	LDP	96(R1), (R15, R16)
+	LDP	112(R0), (R5, R6)
+	LDP	112(R1), (R17, R19)
+	ADCS	R15, R3
+	ADCS	R16, R4
+	STP	(R3, R4), 96(R2)
+	ADCS	R17, R5
+	ADCS	R19, R6
+	STP	(R5, R6), 112(R2)
+
+	LDP	128(R0), (R7, R8)
+	LDP	128(R1), (R15, R16)
+	LDP	144(R0), (R9, R10)
+	LDP	144(R1), (R17, R19)
+	ADCS	R15, R7
+	ADCS	R16, R8
+	STP	(R7, R8), 128(R2)
+	ADCS	R17, R9
+	ADCS	R19, R10
+	STP	(R9, R10), 144(R2)
+
+	LDP	160(R0), (R11, R12)
+	LDP	160(R1), (R15, R16)
+	LDP	176(R0), (R13, R14)
+	LDP	176(R1), (R17, R19)
+	ADCS	R15, R11
+	ADCS	R16, R12
+	STP	(R11, R12), 160(R2)
+	ADCS	R17, R13
+	ADC	R19, R14
+	STP	(R13, R14), 176(R2)
+
+	RET
+
+TEXT ·fp751X2SubLazy(SB), NOSPLIT, $0-24
+	MOVD	z+0(FP), R2
+	MOVD	x+8(FP), R0
+	MOVD	y+16(FP), R1
+
+	LDP	0(R0), (R3, R4)
+	LDP	0(R1), (R15, R16)
+	LDP	16(R0), (R5, R6)
+	LDP	16(R1), (R17, R19)
+	SUBS	R15, R3
+	SBCS	R16, R4
+	STP	(R3, R4), 0(R2)
+	SBCS	R17, R5
+	SBCS	R19, R6
+	STP	(R5, R6), 16(R2)
+
+	LDP	32(R0), (R7, R8)
+	LDP	32(R1), (R15, R16)
+	LDP	48(R0), (R9, R10)
+	LDP	48(R1), (R17, R19)
+	SBCS	R15, R7
+	SBCS	R16, R8
+	STP	(R7, R8), 32(R2)
+	SBCS	R17, R9
+	SBCS	R19, R10
+	STP	(R9, R10), 48(R2)
+
+	LDP	64(R0), (R11, R12)
+	LDP	64(R1), (R15, R16)
+	LDP	80(R0), (R13, R14)
+	LDP	80(R1), (R17, R19)
+	SBCS	R15, R11
+	SBCS	R16, R12
+	STP	(R11, R12), 64(R2)
+	SBCS	R17, R13
+	SBCS	R19, R14
+	STP	(R13, R14), 80(R2)
+
+	LDP	96(R0), (R3, R4)
+	LDP	96(R1), (R15, R16)
+	LDP	112(R0), (R5, R6)
+	LDP	112(R1), (R17, R19)
+	SBCS	R15, R3
+	SBCS	R16, R4
+	SBCS	R17, R5
+	SBCS	R19, R6
+
+	LDP	128(R0), (R7, R8)
+	LDP	128(R1), (R15, R16)
+	LDP	144(R0), (R9, R10)
+	LDP	144(R1), (R17, R19)
+	SBCS	R15, R7
+	SBCS	R16, R8
+	SBCS	R17, R9
+	SBCS	R19, R10
+
+	LDP	160(R0), (R11, R12)
+	LDP	160(R1), (R15, R16)
+	LDP	176(R0), (R13, R14)
+	LDP	176(R1), (R17, R19)
+	SBCS	R15, R11
+	SBCS	R16, R12
+	SBCS	R17, R13
+	SBCS	R19, R14
+	SBC	ZR, ZR, R15
+
+	// Add p751 back but anded with the mask in R15
+	MOVD	·p751+0(SB), R20
+	AND	R15, R20
+	LDP	·p751+40(SB), (R16, R17)
+	ADDS	R20, R3
+	ADCS	R20, R4
+	STP	(R3, R4), 96(R2)
+	ADCS	R20, R5
+	ADCS	R20, R6
+	STP	(R5, R6), 112(R2)
+	ADCS	R20, R7
+
+	LDP	·p751+56(SB), (R19, R20)
+	AND	R15, R16
+	AND 	R15, R17
+	ADCS	R16, R8
+	STP	(R7, R8), 128(R2)
+	ADCS	R17, R9
+
+	LDP	·p751+72(SB), (R16, R17)
+	AND	R15, R19
+	AND	R15, R20
+	ADCS	R19, R10
+	STP	(R9, R10), 144(R2)
+	ADCS	R20, R11
+
+	MOVD	·p751+88(SB), R19
+	AND 	R15, R16
+	AND 	R15, R17
+	ADCS	R16, R12
+	STP	(R11, R12), 160(R2)
+	ADCS	R17, R13
+
+	AND	R15, R19
+	ADC	R19, R14
+	STP	(R13, R14), 176(R2)
+
+	RET
+
+TEXT ·fp751Mul(SB), $0-24
```

Comment:
Can you put comment saying that it is using variant of comba meth.?
More comments in general would be useful